mobile/components/Inputs/CheckboxInput.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { useState } from "react";
+import { Checkbox, Label, XStack, CheckboxProps, CheckedState } from "tamagui";
+import { Icon } from "../Icon";
+import { Typography } from "../Typography";
+
+interface CheckboxInputProps extends CheckboxProps {
+  id: string;
+  label: string;
+}
+
+const CheckboxInput: React.FC<CheckboxInputProps> = ({ label, id, ...rest }) => {
+  const [isChecked, setIsChecked] = useState<CheckedState>(false);
+  return (
+    <XStack
+      height={42}
+      alignItems="center"
+      borderWidth={1}
+      borderColor={isChecked ? "$purple5" : "$gray3"}
+      gap="$xs"
+      paddingHorizontal={14}
+      paddingVertical="$xs"
+      borderRadius={8}
+      {...rest}
+    >
+      <Checkbox
+        width="$md"
+        height="$md"
+        borderRadius={4}
+        id={id}
+        onCheckedChange={(checked) => setIsChecked(checked)}
+        backgroundColor={isChecked ? "$purple5" : "white"}
```

Comment:
Just a curiosity, does Tamagui offer a solution for this checked by overwriting the class with styled()?

Reply:
Nope, cannot control checked state with styled()